File: tinhoctre/file C++8 NGUYENHONGTHAI/1054.py
names = ["vaporeon", "jolteon", "flareon", "espeon", "umbreon", "leafeon", "glaceon", "sylveon"]

n = int(input())
s = input()

for name in names:
    if len(name) != n:
        continue
    match = True
    for i in range(n):
        if s[i] != '.' and s[i] != name[i]:
            match = False
            break
    if match:
        print(name)
        break

File: tinhoctre/file C++8 NGUYENHONGTHAI/1054.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr); cout.tie(nullptr);
    int n; cin >> n;
    string s; cin >> s;
    
    vector<string> names = {
        "vaporeon", "jolteon", "flareon", "espeon",
        "umbreon", "leafeon", "glaceon", "sylveon"
    };
    for(auto& name : names){
        if(name.size() != n){
            continue;
        }
        bool match = true;
        for(int i = 0;i < n; i++){
            if(s[i] != '.' && s[i] != name[i]){
                match = false;
                break;
            }
        }
        if(match){
            cout << name;
            break;
        }
    }
}
